packages/core/test/financial-templates/perpetual-multiparty/PerpetualPositionManager.js

```diff
@@ -1694,6 +1694,39 @@ contract("PerpetualPositionManager", function(accounts) {
     assert(await didContractThrow(positionManager.redeem({ rawValue: "16" }, { from: sponsor })));
   });
 
+  it("Gulp edge cases", async function() {
+    // Gulp does not revert if PfC and collateral balance are both 0
+    await positionManager.gulp();
+
+    // Send 1 wei of excess collateral to the contract.
+    await collateral.transfer(positionManager.address, "1", { from: sponsor });
+
+    // Gulp reverts if PfC is 0 but collateral balance is > 0.
+    assert(await didContractThrow(positionManager.gulp()));
```

Comment:
Just curious what was the behavior before you reverted with 0 pfc? Also this should probably be duplicated for the EMP, i didnt see this check happening.

Reply:
I test this [here](https://github.com/UMAprotocol/protocol/pull/2293/files#diff-b18cec4947547731c8f1cec2b5b452e46e1967d11f8d2879033e94162d913be5R2378). What do you mean by "the behavior before you reverted with 0 pfc"? When PfC and collateral balance are both 0, there is no revert.

Reply:
Ok thanks, yea I thought you explicitly added a new revert condition and was testing it here, but based on your comments it seems you are just adding a new test case. So disregard that question.